torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,271 @@
+                    entry and returns a transformed version
+                cache: if True, cache file handle to speed up repeated sampling
+                download: if True, download dataset and store it in the root directory
+                checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+            Raises:
+                RuntimeError: if ``download=False`` and data is not found, or checksums
+                    don't match
+            """
+        self.root = root
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+        super().__init__(root, crs=crs, res=res, transforms=transforms, cache=cache)
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the extracted files already exist
+        pathname = os.path.join(self.root, "**", self.filename_glob)
+        for fname in glob.iglob(pathname, recursive=True):
+            if not fname.endswith(".zip"):
+                return
+
+        # Check if the zip files have already been downloaded
+        pathname = os.path.join(self.root, self.tarfile_glob)
+        if glob.glob(pathname):
+            self._extract()
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+        # Download the dataset
+        self._download()
+        self._extract()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        for biome, md5 in self.md5s:
+            download_url(
+                self.url.format(biome), self.root, md5=md5 if self.checksum else None
+            )
+
+    def _extract(self) -> None:
+        """Extract the dataset."""
+        pathname = os.path.join(self.root, self.tarfile_glob)
+        for tarfile in glob.iglob(pathname):
+            extract_archive(tarfile)
+    
+    # from LandcoverAIGeo __getitem__
+    def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
+        """Retrieve image/mask and metadata indexed by query.
+
+        Args:
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+
+        Returns:
+            sample of image, mask and metadata at that index
+
+        Raises:
+            IndexError: if query is not found in the index
+        """
+        hits = self.index.intersection(tuple(query), objects=True)
+        img_filepaths = cast(List[str], [hit.object for hit in hits])
+        mask_filepaths = [path.replace(path.split("/")[-1], path.split("/")[-2]+"_mask2019.TIF") for path in img_filepaths]
```

Comment:
Might be easier to follow this logic if you convert the list comprehension to a for-loop. Also, never use `"/"` since that won't work on Windows. Instead use `os.path.split(path)`.